packages/runtime-core/src/componentProps.ts

```diff
@@ -466,7 +466,7 @@ function validateProps(props: Data, instance: ComponentInternalInstance) {
   for (const key in options) {
     let opt = options[key]
     if (opt == null) continue
-    validateProp(key, rawValues[key], opt, !hasOwn(rawValues, key))
+    validateProp(key, rawValues[key], opt, rawValues, !hasOwn(rawValues, key))
```

Comment:
I think this should be cloned deep. Is there an existing utility for that? I could not find one.

Reply:
Also should the prop value we're originally testing for be filtered out of this `rawValues`? (just so there's no confusion why it's available twice)

Reply:
That's perfect. The following still stands:
 > Also should the prop value we're originally testing for be filtered out of this rawValues? (just so there's no confusion why it's available twice)